app/models/manageiq/providers/hawkular/inventory/parser/middleware_manager.rb

```diff
@@ -95,6 +92,8 @@ def fetch_domain_servers(feed)
           server = persister.middleware_servers.find_or_build(domain_server.path)
           parse_middleware_server(domain_server, server, true, server_name)
 
+          associate_with_rhevm(server, feed)
```

Comment:
I migt be missing here something, but is this matching using the same feed for all the servers?